src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -498,12 +474,19 @@ internal String ToString(TraceFormat traceFormat)
 
                     sb.AppendFormat(CultureInfo.InvariantCulture, "   {0} ", word_At);
 
-                    Type t = mb.DeclaringType;
+                    Type declaringType = mb.DeclaringType;
+                    bool isAsync = false;
                     // if there is a type (non global method) print it
-                    if (t != null)
+                    if (declaringType != null && declaringType.IsDefined(typeof(CompilerGeneratedAttribute)) &&
+                        (typeof(IAsyncStateMachine).IsAssignableFrom(declaringType)))
+                    {
+                        isAsync = true;
```

Comment:
Could you add some perf numbers? I had a vague memory you already benchmarked and it was barely noticeable but just want to confirm. Some customers use stack traces on their hot path and get very sad when we slow them down (https://github.com/Microsoft/dotnet/issues/529)

In this case isAsync only needs to be calculated once we determine IsLastFrameFromForeignException is true, but I know if your other use case you need to check it for every frame anyways.

Reply:
Hmm... I tested against throwing the exception and it was insignificant  - but I didn't look at just creating a StackTrace - however this is in the `.ToString` path so wouldn't impact the linked example. 

Never the less, will check